lib/board.rb

```diff
@@ -0,0 +1,54 @@
+#data and logic of the Tic Tac Toe game board
+#board state held in @cells(Array)
+
+class Board
+  attr_accessor :cells
+
+  def initialize
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "] #holds the current state of the game @cells
+  end
+
+  def reset!
+   self.cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "] #clears all tokens from the game @cells
+  end
+
+  def display
+    puts " #{@cells[0]} " + "|" + " #{@cells[1]} " + "|" + " #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} " + "|" + " #{@cells[4]} " + "|" + " #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} " + "|" + " #{@cells[7]} " + "|" + " #{@cells[8]} "
+  end
+
+  def position(pos_from_user)
+    position = pos_from_user.to_i - 1 #decreases num from user to match array indices
+    @cells[position] #returns value from position
+  end
+
+  def full?
```

Comment:
Use the most expressive and efficient iterator for a problem. `Board#full?` is asking if all elements are not empty strings. You can use logic that is symmetrical to those semantics with the [`#all?`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-all-3F) iterator.

`cells.all? {|cell| cell != " "}`

This is also more efficient in that the loop will stop after the iteration for which a cell is not empty.